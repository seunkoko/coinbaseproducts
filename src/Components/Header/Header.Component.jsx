import React from 'react';
import './Header.css';

const Header = () => (
  <nav className="navbar navbar-expand-lg nav__header">
    <div className="contianer header___container">
      <p className="donut__header">tryDONUT</p>
      <p className="coinbase__header">Coinbase Products</p>
    </div>
  </nav>
);

export default Header;
