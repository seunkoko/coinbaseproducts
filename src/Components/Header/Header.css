.nav__header {
  background: #222222;
  color: #D6D5D5;
  height: 60px;
  font-style: normal;
  font-weight: bold;
  line-height: normal;
  padding: 10px;
}

.coinbase__header, .donut__header {
  display: inline;
}

.coinbase__header {
  float: right;
}

.header___container {
  margin: auto;
  margin-top: 10px;
  width: 70%;
}
