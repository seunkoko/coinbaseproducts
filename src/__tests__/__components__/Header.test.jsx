import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import Header from '../../components/Header/Header.Component';

describe('Header', () => {

  it('renders the Header properly', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper.find('nav').length).toEqual(1);
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('p').length).toEqual(2);
    expect(wrapper.find('.nav__header').length).toEqual(1);
    expect(wrapper.find('.donut__header').length).toEqual(1);
    expect(wrapper.find('.coinbase__header').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
});
