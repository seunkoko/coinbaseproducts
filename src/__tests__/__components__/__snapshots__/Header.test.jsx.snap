// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the Header properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="contianer header___container"
      >
        <p
          className="donut__header"
        >
          tryDONUT
        </p>
        <p
          className="coinbase__header"
        >
          Coinbase Products
        </p>
      </div>,
      "className": "navbar navbar-expand-lg nav__header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="donut__header"
          >
            tryDONUT
          </p>,
          <p
            className="coinbase__header"
          >
            Coinbase Products
          </p>,
        ],
        "className": "contianer header___container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "tryDONUT",
            "className": "donut__header",
          },
          "ref": null,
          "rendered": "tryDONUT",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Coinbase Products",
            "className": "coinbase__header",
          },
          "ref": null,
          "rendered": "Coinbase Products",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="contianer header___container"
        >
          <p
            className="donut__header"
          >
            tryDONUT
          </p>
          <p
            className="coinbase__header"
          >
            Coinbase Products
          </p>
        </div>,
        "className": "navbar navbar-expand-lg nav__header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="donut__header"
            >
              tryDONUT
            </p>,
            <p
              className="coinbase__header"
            >
              Coinbase Products
            </p>,
          ],
          "className": "contianer header___container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "tryDONUT",
              "className": "donut__header",
            },
            "ref": null,
            "rendered": "tryDONUT",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Coinbase Products",
              "className": "coinbase__header",
            },
            "ref": null,
            "rendered": "Coinbase Products",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
